pkg/cli/mirror/catalog_images.go

```diff
@@ -88,29 +214,55 @@ func (o *MirrorOptions) rebuildCatalogs(ctx context.Context, dstDir string) (ima
 			if err != nil {
 				return fmt.Errorf("error parsing index dir path %q as image %q: %v", fpath, img, err)
 			}
-			ctlgRef.Ref = sourceRef.Ref
+
+			destRef := image.TypedImage{}
+			destRef.Type = imagesource.DestinationRegistry
+			destRef.Ref = sourceRef.Ref
 			// Update registry so the existing catalog image can be pulled.
-			ctlgRef.Ref.Registry = mirrorRef.Ref.Registry
-			ctlgRef.Ref.Namespace = path.Join(o.UserNamespace, ctlgRef.Ref.Namespace)
-			ctlgRef = ctlgRef.SetDefaults()
+			destRef.Ref.Registry = mirrorRef.Ref.Registry
+			destRef.Ref.Namespace = path.Join(o.UserNamespace, destRef.Ref.Namespace)
+			destRef = destRef.SetDefaults()
 			// Unset the ID when passing to the image builder.
 			// Tags are needed here since the digest will be recalculated.
-			ctlgRef.Ref.ID = ""
+			destRef.Ref.ID = ""
+
+			// associate an imageInfo struct with the destination reference
+			var info imageInfo
+			if existingImageInfo, ok := catalogsByImage[destRef]; ok {
+				info = existingImageInfo
+			} else {
+				info = imageInfo{}
+			}
+			// reconstruct the platform from string and use that platform to lookup the matching platform image and get its hash
+			platform := *NewOperatorCatalogPlatform(platformString)
+			hash, err := getDigestFromOCILayout(ctx, layout.Path(filepath.Join(fullPathToRepoDir, config.LayoutsDir)), platform)
+			if err != nil {
+				return err
+			}
+			info.fullPathToRepoDir = fullPathToRepoDir
+			info.imageDetails = append(info.imageDetails, imageDetails{
+				indexJsonPath: fpath,
+				platform:      platform,
+				hash:          *hash,
```

Comment:
This line is failing in the e2e tests

```goroutine 1 [running]:
github.com/openshift/oc-mirror/pkg/cli/mirror.(*MirrorOptions).rebuildCatalogs.func2({0xc0003e1b00, 0x86}, {0x45656e0, 0xc000cc41a0?}, {0x0?, 0x0})
	/home/lzuccarelli/go/src/github.com/openshift/oc-mirror/pkg/cli/mirror/catalog_images.go:246 +0xb38
path/filepath.walk({0xc0003e1b00, 0x86}, {0x45656e0, 0xc000cc41a0}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:433 +0x123
path/filepath.walk({0xc00044be80, 0x7b}, {0x45656e0, 0xc000cc40d0}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.walk({0xc00044bd00, 0x75}, {0x45656e0, 0xc00099d2b0}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.walk({0xc0005bd180, 0x61}, {0x45656e0, 0xc00099d1e0}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.walk({0xc000c8a060, 0x53}, {0x45656e0, 0xc00099d110}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.walk({0xc0005b2460, 0x49}, {0x45656e0, 0xc00099d040}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.walk({0xc000db64b0, 0x2e}, {0x45656e0, 0xc00099cea0}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.walk({0xc0009a02a0, 0x25}, {0x45656e0, 0xc00099cdd0}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:457 +0x285
path/filepath.Walk({0xc0009a02a0, 0x25}, 0xc00114e3f8)
	/usr/local/go/src/path/filepath/path.go:520 +0x6c
github.com/openshift/oc-mirror/pkg/cli/mirror.(*MirrorOptions).rebuildCatalogs(0xc000288b60, {0x455d250, 0xc0000560e0}, {0xc0009a02a0, 0x25})
	/home/lzuccarelli/go/src/github.com/openshift/oc-mirror/pkg/cli/mirror/catalog_images.go:131 +0x2b8```

Reply:
I see the issue... single arch images won't have a platform associated with the image, and this code needs to handle this situation so it conditionally uses a matcher depending on if a hash was found. I've fixed the code and added a test. Since I can't be sure what e2e test was run here, I am going to leave this conversation open for now.